File: app/src/main/java/workshop1024/com/xproject/model/news/News.java
package workshop1024.com.xproject.model.news;

/**
 * 发布者发布的新闻
 */
public class News {
    //新闻id
    private String mNewId;
    //新闻头图Url
    private String mBannerUrl;
    //新闻标题
    private String mTitle;
    //新闻发布者
    private String mPublisher;
    //新闻发布时间
    private String mPubDate;
    //新闻是否已阅读
    private boolean mIsReaded;
    //新闻是否已保存
    private boolean mIsSaved;

    public News(String newId, String bannerUrl, String title, String publisher, String pubDate,
                boolean isReaded, boolean isSaved) {
        mNewId = newId;
        mBannerUrl = bannerUrl;
        mTitle = title;
        mPublisher = publisher;
        mPubDate = pubDate;
        mIsReaded = isReaded;
        mIsSaved = isSaved;
    }

    public String getNewId() {
        return mNewId;
    }

    public void setNewId(String newId) {
        mNewId = newId;
    }

    public String getBannerUrl() {
        return mBannerUrl;
    }

    public void setBannerUrl(String bannerUrl) {
        mBannerUrl = bannerUrl;
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        mTitle = title;
    }

    public String getPublisher() {
        return mPublisher;
    }

    public void setPublisher(String publisher) {
        mPublisher = publisher;
    }

    public String getPubDate() {
        return mPubDate;
    }

    public void setPubDate(String pubDate) {
        mPubDate = pubDate;
    }

    public boolean isIsReaded() {
        return mIsReaded;
    }

    public void setIsReaded(boolean isReaded) {
        mIsReaded = isReaded;
    }

    public boolean isIsSaved() {
        return mIsSaved;
    }

    public void setIsSaved(boolean isSaved) {
        mIsSaved = isSaved;
    }
}

File: news/src/main/java/workshop1024/com/xproject/news/model/news/News.kt
package workshop1024.com.xproject.news.model.news

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * 发布者发布的新闻
 */
@Entity(tableName = "newses")
data class News(//新闻id
        @PrimaryKey @ColumnInfo(name = "saveId") var mSaveId: String,//内存和本地存储News的id
        @ColumnInfo(name = "newsId") var mNewsId: String, //新闻属于的发布者
        @ColumnInfo(name = "subscribeId") var mSubscribeId: String,//新闻属于的过滤器id
        @ColumnInfo(name = "filterIdList") var mFilterIdList: List<String>,//新闻属于tag的id
        @ColumnInfo(name = "tagIdList") var mTagIdList: List<String>,//新闻头图Url
        @ColumnInfo(name = "bannerUrl") var mBannerUrl: String, //新闻标题
        @ColumnInfo(name = "title") var mTitle: String, //新闻发布者
        @ColumnInfo(name = "publisher") var mPublisher: String, //新闻发布时间
        @ColumnInfo(name = "pubDate") var mPubDate: String,//新闻是否已阅读
        @ColumnInfo(name = "isReaded") var mIsReaded: Boolean, //新闻是否已保存
        @ColumnInfo(name = "isSaved") var mIsSaved: Boolean,//搜索类型
        @ColumnInfo(name = "searchType") var mSearchType: String, //搜索关键字
        @ColumnInfo(name = "searchKey") var mSearchKey: String) {

    companion object {
        const val SUBSCRIBE_TYPE: String = "1"
        const val TAG_TYPE: String = "2"
        const val FILTER_TYPE: String = "3"
        const val SEARCH_TYPE: String = "4"
        const val SAVED_TYPE: String = "5"
    }
}
